src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking_TypeInference.cs

```diff
@@ -1572,6 +1572,9 @@ static void G()
                 references: new[] { ValueTupleRef, SystemRuntimeFacadeRef },
                 parseOptions: TestOptions.Regular8);
             comp.VerifyDiagnostics(
+                // (10,9): warning CS8602: Possible dereference of a null reference.
```

Comment:
:memo: NullableWalker for field access relies on `GetTypeOrReturnTypeWithAdjustedNullableAnnotations` which was modified in this PR. I believe the new behavior is correct.